### PR TITLE
Application for the whole dish psf and eff area

### PR DESCRIPTION
**Pull request by RaulRPrado**
I prepared the application validate_optics which produces PSF and eff area vs off-axis angle using ray tracing simulations. (Tarek asked for the PST for sources at infinity and I decided to do the app at once.) 

@orelgueta, could you please have a look? It's very little code and not urgent.
I called it validate_optics but we can discuss a more specific name if that makes sense.

I did small changes on other apps in order to allow the label to be passed as command line arg.

### CODE REVIEW

File: applications/validate_optics.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import simtools.config as cfg
+import simtools.util.general as gen
+from simtools.model.telescope_model import TelescopeModel
+from simtools.ray_tracing import RayTracing
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(
+        description=(
```
Comment (orelgueta) [inline applications/validate_optics.py:19]: Is this really the description or is it left over from previous code?

File: simtools/ray_tracing.py [hunk 2]
```diff
@@ -350,6 +355,56 @@ def _isValidKeyToPlot(self, key):
...
+        ------
+        KeyError
+            If key is not among the valid options.
+        '''
+        if not self._isValidKeyToPlot(key):
+            msg = 'Invalid key to plot'
+            self._logger.error(msg)
+            raise KeyError(msg)
+
+        ylabel = {
```
Comment (orelgueta) [inline simtools/ray_tracing.py:379]: There is duplication between this list and the list in the _isValidKeyToPlot. I would simply make the dictionary of ylables a class variable and replace the function _isValidKeyToPlot with a simple `if not in ylabel.keys()`.

File: simtools/ray_tracing.py [hunk 3]
```diff
@@ -350,6 +355,56 @@ def _isValidKeyToPlot(self, key):
...
+        self._outputDirectory.joinpath('figures').mkdir(exist_ok=True)
+        plotFile = self._outputDirectory.joinpath('figures').joinpath(plotFileName)
+        self._logger.info('Plotting {} and saving it in {}'.format(key, plotFile))
+
+        plt.figure(figsize=(8, 6), tight_layout=True)
+        ax = plt.gca()
+        ax.set_xlabel('off-axis [deg]')
+        ax.set_ylabel(ylabel[key])
+
+        self.plot(key, **kwargs)
```
Comment (orelgueta) [inline simtools/ray_tracing.py:403]: I can't comment within "plot" cause it wasn't added in this commit, but why not use "visualize" in order to keep all plots with a consistent look?

File: applications/validate_optics.py (hunk 1 above)
Comment (RaulRPrado) [reply applications/validate_optics.py:19]: Left over, sorry.

File: simtools/ray_tracing.py (hunk 2 above)
Comment (RaulRPrado) [reply simtools/ray_tracing.py:379]: Good idea.

File: simtools/ray_tracing.py (hunk 3 above)
Comment (RaulRPrado) [reply simtools/ray_tracing.py:403]: Yes, I agree. I just didn't learn yet to use the visualize, but it's a only matter of spending the time.
I will merge like this and create an issue to remind me to fix that later.

File: simtools/ray_tracing.py [hunk 4]
```diff
@@ -353,7 +361,7 @@ def _isValidKeyToPlot(self, key):
         -------
         bool
         '''
-        return key in ['d80_cm', 'd80_deg', 'eff_area', 'eff_flen']
+        return key in self.YLABEL.keys()
```
Comment (orelgueta) [inline simtools/ray_tracing.py:364]: I would remove this function completely, it's just one command =)

**Pull request merged**